### ISSUE TITLE
Troubles opening ondemand 

### ISSUE THREAD

**Issue by lifetooeasy** [ubccr/hpc-toolset-tutorial#167]
I am trying to run the tutorial on a virtual machine, all containers are running, I can open coldfront and xdmod in the browser. 
But I have problems when trying to open https://<virtualmachinename>:3443, it redirects to localhost:3443.
I did change the ip address in docker-compose.yml but nothing change.

Do you know what should be modified when running in virtual machine? I am not running an extra httpd service.

when running `docker compose logs -f` I see:

```
AH0058: httpd could not reliably determine the servers fully qualified domain name ....
```

**Comment by johrstrom** [comment]
I'm pretty sure you have to keep everything at `localhost` and access through `https://localhost:3443`. Are you saying `https://localhost:3443` doesn't work for you?

**Comment by lifetooeasy** [comment]
yes, I had to change from localhost to 0.0.0.0 (`docker-compose.yml`) this solved the problem with Coldfront and XDMOD but still stuck with ondemand.

**Comment by johrstrom** [comment]
Yea re configuring like that could be troublesome.

We embed all these configs in the container itself. You're going to have to hack these containers or maybe use ssh/X11 forwarding to that you can still use `localhost`.

https://github.com/ubccr/hpc-toolset-tutorial/blob/85c3a8b4b58603a135c84a56516ff8cd9578d107/ondemand/install.sh#L32

**Comment by jgustavsen** [comment]
Hello, I am also having a similar issue, but without any changing of the localhost servername. I can log in to the frontend, to Coldfront, to XDMoD and to the ondemand via ssh but  https://localhost:3443 tells me that the connection is refused.

**Comment by dsajdak** [comment]
@jgustavsen Do you have any other docker containers running that may be using the same port? Is there anything in the ondemand logs

`docker compose logs -f ondemand`

**Comment by jgustavsen** [comment]
Hi @dsajdak ,
Thanks for the quick reply. I restarted the docker and now I am able to access the ondemand login. I did not see any errors in the previous logs, but I will get in touch if I encounter the issue again and can better diagnose.
Thanks, Julia

**Comment by seb45tian** [comment]
Hi, I have the same issue. 

`docker compose logs -f ondemand` show

```
ondemand  | 2023-06-30 09:25:43.098327 I | http: TLS handshake error from 127.0.0.1:35234: local error: tls: bad record MAC
```

And inside the ondemand container in `/var/log/httpd/localhost_error_ssl.log` it says
```
[Fri Jun 30 09:25:43.099922 2023] [auth_openidc:error] [pid 214:tid 139757203998464] [client 172.18.0.1:37458] oidc_util_http_call: curl_easy_perform() failed on: https://localhost:5554/.well-known/openid-configuration (SSL certificate problem: certificate has expired)
[Fri Jun 30 09:25:43.100038 2023] [auth_openidc:error] [pid 214:tid 139757203998464] [client 172.18.0.1:37458] oidc_provider_static_config: could not retrieve metadata from url: https://localhost:5554/.well-known/openid-configuration
```

Is this really an expired certificate? Or are those newly generated when creating the containers? I did do a compleate `./hpcts cleanup ` followed by `./hpcts start` but get the same issue.

**Comment by oberbaum** [comment]
Same issue as @lifetooeasy. Cant connect to ood even if i edit the ood config inside the container and change the servername there to 0.0.0.0.

Any further ideas on that?

**Comment by oberbaum** [comment]
We fixed this issue by changing the IPs of xdmod & coldfront in docker-compose.yml to 0.0.0.0 and a port forwarding for ondemand:
`ssh -L 3443:localhost:3443 -L 5554:localhost:5554 <user@ip>`

**Comment by lifetooeasy** [comment]
@oberbaum tha
